backend/de.metas.payment.sepa/base/src/main/java/de/metas/payment/sepa/sepamarshaller/impl/SEPAVendorCreditTransferMarshaler_Pain_001_001_03_CH_02.java

```diff
@@ -1036,22 +1040,32 @@ private String createInfo(@NonNull final I_SEPA_Export_Line line)
 		{
 			sb.append(" @C_Invoice_ID@ ");
 			sb.append(getInvoiceIDByRecordID(line));
+			sb.append(" @Line@ ");
+			sb.append(getLineNoByRecordID(line));
 		}
 
 		return Services.get(IMsgBL.class).parseTranslation(InterfaceWrapperHelper.getCtx(line), sb.toString());
 	}
 
-	private final IInvoiceDAO invoiceDAO = Services.get(IInvoiceDAO.class);
 	private int getInvoiceIDByRecordID(final I_SEPA_Export_Line sepaline)
 	{
 		final TableRecordReference referencedRecord = TableRecordReference.of(
 				sepaline.getAD_Table_ID(),
 				sepaline.getRecord_ID());
 
-		final I_C_Invoice invoice = invoiceDAO.getByIdInTrx(
-				InvoiceId.ofRepoId(referencedRecord.getRecord_ID()));
+		final I_C_PaySelectionLine paySelectionLine = referencedRecord.getModel(I_C_PaySelectionLine.class);
 
-		return invoice.getC_Invoice_ID();
+		return paySelectionLine.getC_Invoice_ID();
+	}
+
+	private int getLineNoByRecordID(final I_SEPA_Export_Line sepaline)
```

Comment:
This name is misleading because you don't get the line by recordId (SepaExportLineId)  but by record (I_SEPA_Export_Line). Because it's a private method, I would name it simply ``getLineNo``

Reply:
sepaline should be @NonNull , right?